api/submission/remove_submission.go

```diff
@@ -0,0 +1,36 @@
+package submission
+
+import (
+    "github.com/eriq-augustine/autograder/api/core"
+    "github.com/eriq-augustine/autograder/db"
+)
+
+type RemoveSubmissionRequest struct {
+    core.APIRequestAssignmentContext
+    core.MinRoleGrader
+
+    TargetUser core.TargetUserSelfOrGrader `json:"target-email"`
+    TargetSubmission string `json:"target-submission"`
```

Comment:
This typing (being a string that can be empty) indicates that if no ID is supplied the most recent submission is used.
I think that this is the correct functionality, but the rest of your code assumes that a submission ID is always provided.

Reply:
Doesn't the autograder-py interface only send a request to the server if the targetsubmission is not empty, meaning if there was a request there will always be a submissionID since it got through that check in autograder-py?

Reply:
What about when people don't use the autograder-py library to send requests?
Or they use the library, but override/bypass our code?

Regardless, I recommended changing the Python side as well.
This is also a really strange statement, since you are also making the Python side right now.
Even if the other side was not currently in development, we should always strive to do the right thing, even if it means changing standards.